examples/hosts/app-integration/external-data/src/model/taskListCollection.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { SharedMap } from "@fluidframework/map";
+
+import type { ITaskListCollection, ITaskListInitialState } from "../model-interface";
+import { TaskList, TaskListInstantiationFactory } from "./taskList";
+
+export class TaskListCollection extends DataObject implements ITaskListCollection {
+	/**
+	 * The taskLists map holds local facades on the data.
+	 */
+	private readonly taskLists = new Map<string, TaskList>();
+
+	private _taskListCollection: SharedMap | undefined;
```

Comment:
I think this class can be simplified a bit; `this.root` will already be a SharedMap guaranteed to always be there, so we could do away with `_taskListCollection`. And the `tasksLists` facade seems a bit unnecessary? Avoids a bit of `async`-ness but not sure it's worth it, and I'm also not seeing how it gets updated in client A if client B adds a task list; `_taskListCollection` would get updated because it's a fluid shared object, but we're not listening for changes there to update the facade, right?

Reply:
`tasksLists`'s non asyncness is necessary in order for the view to be able to render it. Actually, at the moment, it looks like the rendering doesn't even work for the second client that loads. Let me figure out the full flow.